pkg/app/piped/eventwatcher/eventwatcher.go

```diff
@@ -165,9 +171,21 @@ func (w *watcher) run(ctx context.Context, repo git.Repo, repoCfg config.PipedRe
 	}
 }
 
+func (w *watcher) cloneRepo(ctx context.Context, repoCfg config.PipedRepository) (git.Repo, error) {
+	repo, err := w.gitClient.Clone(ctx, repoCfg.RepoID, repoCfg.Remote, repoCfg.Branch, "")
+	if err != nil {
+		w.logger.Error("failed to clone repository",
+			zap.String("repo-id", repoCfg.RepoID),
+			zap.Error(err),
+		)
+		return nil, fmt.Errorf("failed to clone repository %s: %w", repoCfg.RepoID, err)
+	}
+	return repo, nil
+}
+
 // updateValues inspects all Event-definition and pushes the changes to git repo if there is.
 func (w *watcher) updateValues(ctx context.Context, repo git.Repo, events []config.EventWatcherEvent, commitMsg string) error {
-	// Copy the repo to another directory to avoid pull failure in the future.
+	// Copy the repo to another directory to modify local file to avoid reverting previous changes.
 	tmpDir, err := ioutil.TempDir("", "event-watcher")
```

Comment:
Previously, to avoid pull failure in the future, it copied the repository.
Ref: https://github.com/pipe-cd/pipe/pull/1411#discussion_r557894485

However, we should assume that the commit history of the repository will be broken for some reason kind of like the possibility of force pushes and so on.